GoogleForCreators/web-stories-wp
# Video Optimization: First Time Usage Dialog

## Context



## Summary

By default, video optimization is set to opt in. When the user encounters this feature for the first time, they will see a dialog with an explanation.

The dialog is displayed while optimization is already in prgoress, so it does not pause/stop the process or anything.

Note: "For the first time" means this is recorded in localStorage, not user meta.


Required https://github.com/google/web-stories-wp/pull/6429 to be merged first. 

## Relevant Technical Choices

Exposing `isTranscoding` in media provider, forces the re-render that is required. 
Also check if `isTranscodingEnabled` in user meta as well, just incase the user has opted out in settings. 

## To-do



## User-facing changes

<img width="1884" alt="Screenshot 2021-02-19 at 17 49 58" src="https://user-images.githubusercontent.com/237508/108541931-fa238180-72da-11eb-8b7b-60bb618596cb.png">

## Testing Instructions



### QA

This PR can be tested by following these steps:

1. Enable video transcoding experiment
2. Open editor. 
3. Drag a video file into editor. 
4. See dialog.  

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6087
